core/platform/storage/cloud_storage_emulator.py

```diff
@@ -0,0 +1,188 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""An emulator that mocks the core.platform.storage API."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals # pylint: disable=import-only-modules
+
+import pickle
+
+import feconf
+import python_utils
+
+import redis
+
+REDIS_CLIENT = redis.StrictRedis(
+    host=feconf.REDISHOST,
+    port=feconf.REDISPORT,
+    db=feconf.STORAGE_EMULATOR_REDIS_DB
+)
+
+
+class Blob(python_utils.OBJECT):
```

Comment:
I want to preserve more direct mapping to the Cloud Storage.

Reply:
I'm a bit confused about this, actually. This file has a replacement for the cloud storage emulator. But why can't we just use the existing Blob class in cloud storage rather than rewriting our own?

Reply:
Because the existing Blob class is not just basic domain class (data holder), but id does opperations on the Cloud Storage to, so there is no simple way to use it.

Reply:
Are we preserving full interoperability between these two Blob classes then? If not, at least call this EmulatorBlob or something like that so that we don't mistakenly operate on one when we mean the other. They need to be regarded as completely different things.

Reply:
Yes, renaming makes sense. Done.